File: app/src/main/java/pe/ebim/canchaslima/ui/screens/LoginScreen.kt
package pe.ebim.canchaslima.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import pe.ebim.canchaslima.data.SampleData

@Composable
fun LoginScreen(onIngresar: (String) -> Unit) {
    var club by remember { mutableStateOf(SampleData.CLUB_ACTIVO) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(28.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text("🎾", style = MaterialTheme.typography.displayMedium)
        Spacer(Modifier.height(8.dp))
        Text(
            "Canchas Lima",
            style = MaterialTheme.typography.headlineMedium,
            fontWeight = FontWeight.Bold,
            color = MaterialTheme.colorScheme.primary
        )
        Text(
            "Panel del Club · Llena tus horas vacías",
            style = MaterialTheme.typography.bodyMedium,
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
        Spacer(Modifier.height(40.dp))

        OutlinedTextField(
            value = club,
            onValueChange = { club = it },
            label = { Text("Nombre del club") },
            singleLine = true,
            modifier = Modifier.fillMaxWidth()
        )
        Spacer(Modifier.height(16.dp))
        Button(
            onClick = { onIngresar(club) },
            modifier = Modifier.fillMaxWidth()
        ) {
            Text("Ingresar al panel")
        }
        Spacer(Modifier.height(24.dp))
        Text(
            "Acceso de demostración · Fase 1 (captación de oferta)",
            style = MaterialTheme.typography.labelSmall,
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
    }
}

File: lib/screens/login_screen.dart
import 'package:flutter/material.dart';

import '../brand.dart';
import '../data/sample_data.dart';
import '../state/app_state.dart';
import '../theme.dart';
import 'home_shell.dart';

class LoginScreen extends StatefulWidget {
  const LoginScreen({super.key});

  @override
  State<LoginScreen> createState() => _LoginScreenState();
}

class _LoginScreenState extends State<LoginScreen> {
  late final TextEditingController _ctrl =
      TextEditingController(text: SampleData.clubActivo);

  @override
  void dispose() {
    _ctrl.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Colors.transparent,
        elevation: 0,
        foregroundColor: verdeCancha,
        title: const Text('Soy dueño de cancha',
            style: TextStyle(color: verdeCancha, fontSize: 16)),
      ),
      extendBodyBehindAppBar: true,
      body: SafeArea(
        child: Center(
          child: SingleChildScrollView(
            padding: const EdgeInsets.all(28),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                const Text('🎾', style: TextStyle(fontSize: 56)),
                const SizedBox(height: 8),
                Text(
                  kBrandName,
                  style: Theme.of(context)
                      .textTheme
                      .headlineMedium
                      ?.copyWith(fontWeight: FontWeight.bold, color: verdeCancha),
                ),
                Text(
                  'Panel del Club · Llena tus horas vacías',
                  textAlign: TextAlign.center,
                  style: Theme.of(context).textTheme.bodyMedium,
                ),
                const SizedBox(height: 40),
                TextField(
                  controller: _ctrl,
                  decoration: const InputDecoration(
                    labelText: 'Nombre del club',
                    border: OutlineInputBorder(),
                  ),
                ),
                const SizedBox(height: 16),
                SizedBox(
                  width: double.infinity,
                  child: FilledButton(
                    onPressed: () {
                      appState.iniciarSesion(_ctrl.text);
                      Navigator.of(context).pushReplacement(
                        MaterialPageRoute(builder: (_) => const HomeShell()),
                      );
                    },
                    style: FilledButton.styleFrom(
                      backgroundColor: verdeCancha,
                      padding: const EdgeInsets.symmetric(vertical: 14),
                    ),
                    child: const Text('Ingresar al panel'),
                  ),
                ),
                const SizedBox(height: 24),
                Text(
                  'Acceso de demostración · Fase 1 (captación de oferta)',
                  textAlign: TextAlign.center,
                  style: Theme.of(context).textTheme.labelSmall,
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
